### PR TITLE
allow formatting options

### PR DESCRIPTION
**Pull request by aviatesk**
Will work with: https://github.com/JunoLab/Atom.jl/pull/259

### CODE REVIEW

**Comment by aviatesk** [comment]
Rebase after https://github.com/JunoLab/atom-julia-client/pull/677 gets merged

File: lib/package/config.coffee [hunk 1]
```diff
@@ -105,6 +106,75 @@ config =
...
         default: false
         order: 13
+      formattettingOptions:
+        title: 'Formatting options'
+        description: 'Specifies [formatting options for `JuliaFormatter.format_text`](https://domluna.github.io/JuliaFormatter.jl/stable/#Formatting-Options-1).'
+        type: 'object'
+        order: 14
+        collapsed: true
+        properties:
+          indent:
```
Comment (pfitzseb) [inline lib/package/config.coffee]: I don't think we need that option -- it's intuitive to have JuliaFormatter follow the current editor, no?

File: lib/package/config.coffee [hunk 2]
```diff
@@ -105,6 +106,75 @@ config =
...
+            title: 'indent'
+            description:
+              '''
+              Specifies the number of spaces used for an indentation.
+              With the default value (`-1`), `indent` will be automatically set to the current editor's [tab length](https://flight-manual.atom.io/using-atom/sections/basic-customization/#configuration-key-reference).
+              '''
+            type: 'integer'
+            default: -1
+            order: 1
+          margin:
```
Comment (pfitzseb) [inline lib/package/config.coffee]: Same here, but maybe to a lesser extent.

File: lib/package/config.coffee (hunk 1 above)
Comment (aviatesk) [reply lib/package/config.coffee]: It's very intuitive that `indent` and `margin` follow the current editor's ones, but why I added them is that "some" project may want members to have the same setting of `indent` and `margin` for formatting within a project, in the future.
Those `indent` and `margin` may be different from those an user have specified as their general (or language specific) setting.
What I'm thinking about is somewhat will be enabled by https://github.com/domluna/JuliaFormatter.jl/issues/184.

***

But at the same time, I feel those settings are kinda too much for now -- I don't think currently there is a Julia project that encourages members to use the exactly same setting even for local formatting task.

So as a conclusion I'm totally okay to remove these configs for now.

File: lib/package/config.coffee [hunk 3]
```diff
@@ -105,6 +106,75 @@ config =
         type: 'boolean'
         default: false
         order: 13
+      formattettingOptions:
```
Comment (waldyrious) [inline lib/package/config.coffee]:
```suggestion
      formattingOptions:
```

Comment (aviatesk) [reply lib/package/config.coffee]: thanks :)

File: lib/package/config.coffee (hunk 1 above)
Comment (pfitzseb) [reply lib/package/config.coffee]: Let's keep this for now then.

**Pull request merged**